src/workerd/api/r2-test.js

```diff
@@ -107,6 +107,15 @@ async function compareResponse(res, { head, body } = {}) {
     ...head,
   });
   if (body) {
+    if (bytes) {
+      const expected = new Uint8Array(new TextEncoder().encode(body));
+      const actual = await (await res).bytes();
+      assert.equal(expected.byteLength, actual.byteLength);
+      for (let i = 0; i < expected.byteLength; i++) {
+        assert.equal(expected[i], actual[i]);
+      }
+      return;
+    }
     assert.strictEqual(await (await res).text(), body);
```

Comment:
Should this statement be in an `else` block? Doesn't calling `(await res).bytes()` drain the body so that calling `(await res).text()` will fail?

Reply:
Doh! I missed the `return` statement. Cool